cli/server/server.go

```diff
@@ -259,24 +262,43 @@ func restoreDB(ctx *cli.Context) error {
 		chain.Close()
 	}()
 
-	var start uint32
-	if ctx.Bool("incremental") {
-		start = reader.ReadU32LE()
-		if chain.BlockHeight()+1 < start {
-			return cli.Exit(fmt.Errorf("expected height: %d, dump starts at %d",
-				chain.BlockHeight()+1, start), 1)
-		}
+	var start = reader.ReadU32LE()
+	if reader.Err != nil {
+		return cli.Exit(err, 1)
+	}
+	var allBlocks = reader.ReadU32LE()
+	if reader.Err != nil {
+		return cli.Exit(err, 1)
+	}
+	var versionOrLen = reader.ReadU32LE()
+	if reader.Err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Block's len cannot be equal to 0 => the block version has been read
+	// => the number of blocks in the file is in start
```

Comment:
godot, and the comment may be shortened and adjusted, we need to mention compatibility with the full dump format.